File: БР1.1/Захматов Юрий/labs/lab2/chat-service/src/main/java/com/petproject/chatservice/security/JwtAuthenticationFilter.java
package com.petproject.chatservice.security;


import jakarta.servlet.FilterChain;
import jakarta.servlet.ServletException;
import jakarta.servlet.http.HttpServletRequest;
import jakarta.servlet.http.HttpServletResponse;
import lombok.NonNull;
import lombok.RequiredArgsConstructor;
import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
import org.springframework.security.core.context.SecurityContextHolder;
import org.springframework.stereotype.Component;
import org.springframework.web.filter.OncePerRequestFilter;

import java.io.IOException;

@Component
@RequiredArgsConstructor
public class JwtAuthenticationFilter extends OncePerRequestFilter {

    private final JwtService jwtService;

    @Override
    protected void doFilterInternal(
            @NonNull HttpServletRequest request,
            @NonNull HttpServletResponse response,
            @NonNull FilterChain filterChain
    ) throws ServletException, IOException {

        final String authorizationHeader = request.getHeader("Authorization");

        if (authorizationHeader == null || !authorizationHeader.startsWith("Bearer ")) {
            filterChain.doFilter(request, response);
            return;
        }

        final String token = authorizationHeader.substring(7);

        if (!jwtService.isValid(token)) {
            response.setStatus(HttpServletResponse.SC_UNAUTHORIZED);
            return;
        }

        Long userId = jwtService.extractUserId(token);
        String username = jwtService.extractUsername(token);
        String role = jwtService.extractRole(token);
        Boolean isLandlord = jwtService.extractIsLandlord(token);
        Boolean isRenter = jwtService.extractIsRenter(token);


        JwtPrincipal principal = JwtPrincipal.builder()
                                             .userId(userId)
                                             .username(username)
                                             .role(role)
                                             .isLandlord(isLandlord)
                                             .isRenter(isRenter)
                                             .build();

        UsernamePasswordAuthenticationToken auth =
                new UsernamePasswordAuthenticationToken(principal, token, principal.getAuthorities());

        SecurityContextHolder.getContext().setAuthentication(auth);

        filterChain.doFilter(request, response);
    }
}

File: БР1.2/Цатинян Артём/labs/lab1/src/main/kotlin/ru/itmo/restaurantbooking/auth/config/JwtAuthenticationFilter.kt
package ru.itmo.restaurantbooking.auth.config

import jakarta.servlet.FilterChain
import jakarta.servlet.http.HttpServletRequest
import jakarta.servlet.http.HttpServletResponse
import org.springframework.security.authentication.UsernamePasswordAuthenticationToken
import org.springframework.security.core.authority.SimpleGrantedAuthority
import org.springframework.security.core.context.SecurityContextHolder
import org.springframework.stereotype.Component
import org.springframework.web.filter.OncePerRequestFilter
import ru.itmo.restaurantbooking.auth.service.JwtService

@Component
class JwtAuthenticationFilter(
    private val jwtService: JwtService
) : OncePerRequestFilter() {

    override fun doFilterInternal(request: HttpServletRequest, response: HttpServletResponse, filterChain: FilterChain) {
        val header = request.getHeader("Authorization")
        if (header != null && header.startsWith("Bearer ")) {
            val token = header.removePrefix("Bearer ").trim()
            try {
                val user = jwtService.parse(token)
                val authorities = listOf(SimpleGrantedAuthority("ROLE_${user.role.name}"))
                SecurityContextHolder.getContext().authentication = UsernamePasswordAuthenticationToken(user, null, authorities)
            } catch (_: Exception) {
            }
        }
        filterChain.doFilter(request, response)
    }
}
